### PR TITLE
Unified workflow result output

### PR DESCRIPTION
**Pull request by Kami**
* Change action chain runner result output matches unified format proposed by @enykeev 
* Change mistral runner result output matches unified format proposed by @enykeev 

/cc #1045

### CODE REVIEW

**Comment by manasdk** [comment]
We are ok with this being a temporary change - right? I am pretty sure that within the next couple of weeks we will break up the roll-up part.

**Comment by Kami** [comment]
@manasdk You mean when we will also want to display sub work-flows?

**Comment by manasdk** [comment]
@kami - Correct to support sub work-flows properly will have to stop rolling up results at any level.

**Comment by Kami** [comment]
@manasdk Won't we still be able to return a list of tasks?

We will just need to add parent_id and child_id to each task in case we want to return flat list instead of nested objects or?

**Comment by manasdk** [comment]
> Won't we still be able to return a list of tasks?
For API I propose we come up with a separate view which does the correct roll-up or atleast presents all the data.

DB-wise. I believe each execution should only refer to children and/or a parent. Each execution object must only store its result and not a list of tasks. 

With this PR we also store task data in an execution and return it via API. I know this PR is basically an extension of what was going on before but I just to make sure that we all agree that this is not how we store data going forward.

**Comment by manasdk** [comment]
*each task* should be represented by its own ActionExecution. In effect we do not have the distinction between ActionExecution and TaskExecution - those are mistral terms which we can abstract away.

**Comment by Kami** [comment]
Thanks for the clarification and I agree about a separate view and that each task should have it's own ActionExecution object.

**Comment by manasdk** [comment]
Perfect. Btw, I am ok with this change as an interim approach. Just wanted to call out that this will break so that @enykeev is aware that things will break.

File: st2actions/st2actions/runners/actionchainrunner.py [hunk 1]
```diff
@@ -142,41 +144,66 @@ def pre_run(self):
...
+                    action_node=action_node, execution_result=actionexec.result,
+                    previous_execution_results=context_result, chain_vars=self.chain_holder.vars)
+
                 if rendered_publish_vars:
                     self.chain_holder.vars.update(rendered_publish_vars)
             finally:
+                # Record result and resolve a next node based on the task success or failure
+                updated_at = datetime.datetime.now()
+
+                format_kwargs = {'action_node': action_node, 'action_exec_db': actionexec,
```
Comment (manasdk) [inline st2actions/st2actions/runners/actionchainrunner.py]: If this comes from the except block action_exec_db will be None. I will fix it on the rename branch.

Comment (manasdk) [inline st2actions/st2actions/runners/actionchainrunner.py]: Nope. I am wrong it is handled properly.

**Pull request merged**